source/javascripts/components/unified-editor/WorkflowConfig/components/EnvVarsCard.tsx

```diff
@@ -82,14 +71,8 @@ const EnvVarCard = ({ id, index, onRemove }: { id: string; index: number; onRemo
             aria-label="Key"
             leftIconName="Dollars"
             placeholder="Enter key"
-            errorText={errors.envs?.[index]?.key?.message}
-            {...register(`envs.${index}.key`, {
-              validate: (v) =>
-                EnvVarService.validateKey(
-                  v,
-                  envVars.map((env) => env.key),
-                ),
-            })}
+            errorText={formState.errors.configuration?.envs?.[index]?.key?.message}
+            {...register(`configuration.envs.${index}.key`, { validate: (v) => EnvVarService.validateKey(v) })}
```

Comment:
Should we stick to one style?
```suggestion
            {...register(`configuration.envs.${index}.key`, { validate: EnvVarService.validateKey })}
```